src/helpers/assertStringsWithinText.ts

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import { ReactTestInstance } from 'react-test-renderer';
+import { filterNodeByType } from './filterNodeByType';
+
+export const assertStringsWithinText = (instance: ReactTestInstance): void => {
+  const nodesWithStringChild = instance.findAll((node) =>
+    isSomeChildStringOrNumber(node.props.children)
+  );
+  nodesWithStringChild.forEach((node) => {
+    const isHostTextComponent = filterNodeByType(node, 'Text');
+    const isNativeComponent = typeof node.type === 'string';
+
+    if (!isHostTextComponent && isNativeComponent) {
+      throw new Error(
```

Comment:
Would it be possible to get a better error, so the user can figure out where is the problem? Overall `@testing-library/react` does a good job with its errors to help the user.

In our case, I would expect at least the string rendered to be logged, ideally we'd get s snapshot of the part of the tree that poses problem (but I'm not sure RNTL offers that possibility)

Reply:
Logging part of the tree might be difficult without having `toJSON` being exported from React Test Renderer. But logging text string should work

Reply:
Yes I think logging the string that is rendered is a good idea, I don't see how we could log a part of the tree though